src/nano_contracts/utils.ts

```diff
@@ -28,24 +27,26 @@ import {
 import { OutputType } from '../wallet/types';
 import { IStorage } from '../types';
 import { parseScript } from '../utils/scripts';
-import { decryptData } from '../utils/crypto';
-import { PrivateKey } from 'bitcore-lib';
 import { MethodArgInfo } from './types';
 
 /**
  * Sign a transaction, create a send transaction object, mine and push
  *
- * @param {tx} Transaction to sign and send
- * @param {privateKey} Private key of the nano contract's tx signature
- * @param {pin} Pin to decrypt data
- * @param {storage} Wallet storage object
+ * @param tx Transaction to sign and send
+ * @param pin Pin to decrypt data
+ * @param storage Wallet storage object
  */
-export const signAndPushNCTransaction = async (tx: NanoContract, privateKey: PrivateKey, pin: string, storage: IStorage): Promise<Transaction> => {
-  const dataToSignHash = tx.getDataToSignHash();
-  // Add nano signature
-  tx.signature = transactionUtils.getSignature(dataToSignHash, privateKey);
-  // Inputs signature, if there are any
-  await transactionUtils.signTransaction(tx, storage, pin);
+export const signAndPushNCTransaction = async (tx: NanoContract, pin: string, storage: IStorage): Promise<Transaction> => {
+  const txSigData = await storage.getTxSignatures(tx, pin);
+  for (const sigData of txSigData.inputSignatures) {
+    const input = tx.inputs[sigData.inputIndex];
+    const inputData = transactionUtils.createInputData(
+      sigData.signature,
+      sigData.pubkey,
+    );
+    input.setData(inputData);
+  }
```

Comment:
You have a method in the `transactionUtils` (`signTransaction`) that has precisely this same loop. What do you think about creating a util method that receives the signatures and tx to reuse the code?

Reply:
Changed the `signTransaction` to include the nano_contract specific logic so we can use it here